// vscode-json-languageservice-minecraft/lib/umd/minecraftServices/minecraftCompletion.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "./minecraftUtils", "vscode-languageserver-types", "./serverEntity"], factory);
    }
})(function (require, exports) {
    'use strict';
    Object.defineProperty(exports, "__esModule", { value: true });
    var minecraftUtils_1 = require("./minecraftUtils");
    var vscode_languageserver_types_1 = require("vscode-languageserver-types");
    var ServerEntity = require("./serverEntity");
    var MinecraftCompletion = /** @class */ (function () {
        function MinecraftCompletion() {
        }
        MinecraftCompletion.prototype.addMinecraftValueCompletions = function (document, schema, separatorAfter, collector) {
            var _this = this;
            if (schema.minecraftType) {
                switch (minecraftUtils_1.getMinecraftDocumentType(document)) {
                    case minecraftUtils_1.MinecraftDocumentType.ServerEntity: {
                        if (schema.minecraftType === "componentGroupName") {
                            ServerEntity.getComponentGroups(document).forEach(function (x) { return _this.addStringValueCompletion(x, separatorAfter, collector); });
                        }
                    }
                }
            }
        };
        MinecraftCompletion.prototype.addStringValueCompletion = function (value, separatorAfter, collector) {
            collector.add({
                kind: vscode_languageserver_types_1.CompletionItemKind.Value,
                label: JSON.stringify(value),
                insertText: JSON.stringify(value) + separatorAfter,
                insertTextFormat: vscode_languageserver_types_1.InsertTextFormat.PlainText
            });
        };
        return MinecraftCompletion;
    }());
    exports.MinecraftCompletion = MinecraftCompletion;
});
