// vscode-json-languageservice-minecraft/lib/esm/minecraftServices/minecraftCompletion.js
'use strict';
import { MinecraftDocumentType, getMinecraftDocumentType } from './minecraftUtils';
import { CompletionItemKind, InsertTextFormat } from 'vscode-languageserver-types';
import * as ServerEntity from './serverEntity';
var MinecraftCompletion = /** @class */ (function () {
    function MinecraftCompletion() {
    }
    MinecraftCompletion.prototype.addMinecraftValueCompletions = function (document, schema, separatorAfter, collector) {
        var _this = this;
        if (schema.minecraftType) {
            switch (getMinecraftDocumentType(document)) {
                case MinecraftDocumentType.ServerEntity: {
                    if (schema.minecraftType === "componentGroupName") {
                        ServerEntity.getComponentGroups(document).forEach(function (x) { return _this.addStringValueCompletion(x, separatorAfter, collector); });
                    }
                }
            }
        }
    };
    MinecraftCompletion.prototype.addStringValueCompletion = function (value, separatorAfter, collector) {
        collector.add({
            kind: CompletionItemKind.Value,
            label: JSON.stringify(value),
            insertText: JSON.stringify(value) + separatorAfter,
            insertTextFormat: InsertTextFormat.PlainText
        });
    };
    return MinecraftCompletion;
}());
export { MinecraftCompletion };
